misc/json.c

```diff
@@ -300,12 +300,15 @@ static int json_append(bstr *b, const struct mpv_node *src, int indent)
         bstr_xappend_asprintf(NULL, b, "%"PRId64, src->u.int64);
         return 0;
     case MPV_FORMAT_DOUBLE: {
-        const char *px = isfinite(src->u.double_) ? "" : "\"";
+        const char *px = (isfinite(src->u.double_) || indent == 0) ? "" : "\"";
         bstr_xappend_asprintf(NULL, b, "%s%f%s", px, src->u.double_, px);
         return 0;
     }
     case MPV_FORMAT_STRING:
-        write_json_str(b, src->u.string);
+        if (indent == 0)
+            APPEND(b, src->u.string);
+        else
+            write_json_str(b, src->u.string);
```

Comment:
This is only used for stringizing properties, and `indent == 0` means this only happens when (a) we're in "print" mode (i.e. pretty-printing, non-raw-mode text expansion), and (b) this is the top-level object. So basically, it means that directly pretty-printing the value of a string node will just give the string content, rather than the escaped/quote-wrapped version. This is substantially more useful for parameter expansion, and still allows using raw mode if you need something guaranteed to be parseable as JSON.

Reply:
If this is only used during pretty printing then that's fine, but I'm pretty sure this code is also used for IPC so it needs to be valid json.

Reply:
`json_write` is used elsewhere (including in IPC), but `json_write_pretty` is used only via `m_option_pretty_print`, which is for human viewing, and the `indent == 0` case only occurs in `json_write_pretty`.